internal/servers/controller/testing.go

```diff
@@ -629,3 +638,63 @@ func (tc *TestController) AddClusterControllerMember(t *testing.T, opts *TestCon
 	}
 	return NewTestController(t, nextOpts)
 }
+
+// WaitForNextWorkerStatusUpdate waits for the next status check from a worker to
+// come in. If it does not come in within the default status grace
+// period, this function returns an error.
+func (tc *TestController) WaitForNextWorkerStatusUpdate(workerId string) error {
+	tc.Logger().Debug("waiting for next status report from worker", "worker", workerId)
+	waitStatusStart := time.Now()
+	ctx, cancel := context.WithTimeout(tc.ctx, tc.b.StatusGracePeriodDuration)
+	defer cancel()
+	var err error
+	for {
+		if err = ctx.Err(); err != nil {
```

Comment:
Not sure if you wanted to implement the same suggestion Jeff gave re turning this into a for select

Reply:
Should be done now!